File: apiwznm/WznmQLibList.cpp
/**
	* \file WznmQLibList.cpp
	* API code for table TblWznmQLibList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 25 Aug 2020
	* \date modified: 25 Aug 2020
	*/

#include "WznmQLibList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQLibList
 ******************************************************************************/

WznmQLibList::WznmQLibList(
			const uint jnum
			, const string sref
			, const string Title
			, const string Version
			, const string srefKLictype
			, const string titSrefKLictype
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Title = Title;
	this->Version = Version;
	this->srefKLictype = srefKLictype;
	this->titSrefKLictype = titSrefKLictype;
};

bool WznmQLibList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQLibList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Version", "ver", Version);
		extractStringUclc(docctx, basexpath, "srefKLictype", "lty", srefKLictype);
		extractStringUclc(docctx, basexpath, "titSrefKLictype", "lty2", titSrefKLictype);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQLibList
 ******************************************************************************/

ListWznmQLibList::ListWznmQLibList() {
};

ListWznmQLibList::ListWznmQLibList(
			const ListWznmQLibList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQLibList(*(src.nodes[i]));
};

ListWznmQLibList& ListWznmQLibList::operator=(
			const ListWznmQLibList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQLibList(*(src.nodes[i]));

	return *this;
};

ListWznmQLibList::~ListWznmQLibList() {
	clear();
};

void ListWznmQLibList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQLibList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQLibList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQLibList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQLibList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQLibList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQLibList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQLibList.java
/**
	* \file WznmQLibList.h
	* Java API code for record of table TblWznmQLibList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQLibList {
	
	public WznmQLibList(
				int jnum
				, String sref
				, String Title
				, String Version
				, String srefKLictype
				, String titSrefKLictype
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Title = Title;
		this.Version = Version;
		this.srefKLictype = srefKLictype;
		this.titSrefKLictype = titSrefKLictype;
	};

	public int jnum;
	public String sref;
	public String Title;
	public String Version;
	public String srefKLictype;
	public String titSrefKLictype;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQLibList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Version = Xmlio.extractStringUclc(doc, basexpath, "Version", "ver", null, 0);
			srefKLictype = Xmlio.extractStringUclc(doc, basexpath, "srefKLictype", "lty", null, 0);
			titSrefKLictype = Xmlio.extractStringUclc(doc, basexpath, "titSrefKLictype", "lty2", null, 0);
			
			return true;
		};

		return false;
	};
};

